activestate.generators.yaml

```diff
@@ -94,25 +94,34 @@ scripts:
     value: |
       set -e
       $constants.SET_ENV
-      mkdir -p ./build/test/state-install/bin
-      mkdir -p ./build/test-update
+      TEST_BASE_DIR=./build/test-payload
+      TEST_STAGED_DIR=${TEST_BASE_DIR}/staged-components
+      TEST_STAGED_BIN_DIR=${TEST_STAGED_DIR}/bin
+      TEST_PREPARED_DIR=${TEST_BASE_DIR}/state-install
+      TEST_ARCHIVE_DIR=./build/test-update
+      mkdir -p ${TEST_STAGED_BIN_DIR}
+      mkdir -p ${TEST_ARCHIVE_DIR}
       TEST_VERSION="99.99.9999"
       TEST_CHANNEL=${1:-test-channel}
 
-      TEST_STATE_TOOL=./build/test/state-install/bin/$constants.BUILD_TARGET
-      TEST_STATE_SVC=./build/test/state-install/bin/$constants.BUILD_DAEMON_TARGET
-      TEST_STATE_EXEC=./build/test/state-install/bin/$constants.BUILD_EXEC_TARGET
-      TEST_INSTALLER=./build/test/state-install/$constants.BUILD_INSTALLER_TARGET
+      TEST_STATE_TOOL=${TEST_STAGED_BIN_DIR}/$constants.BUILD_TARGET
+      TEST_STATE_SVC=${TEST_STAGED_BIN_DIR}/$constants.BUILD_DAEMON_TARGET
+      TEST_STATE_EXEC=${TEST_STAGED_BIN_DIR}$constants.BUILD_EXEC_TARGET
+      TEST_INSTALLER=${TEST_STAGED_DIR}/$constants.BUILD_INSTALLER_TARGET
       go build -o $TEST_STATE_TOOL ./internal/testhelpers/installation/fakeversioncmd
       cp -a $TEST_STATE_TOOL $TEST_STATE_SVC
       cp -a $TEST_STATE_TOOL $TEST_STATE_EXEC
       cp -a $BUILD_TARGET_DIR/$constants.BUILD_INSTALLER_TARGET $TEST_INSTALLER
 
-      go run scripts/ci/update-generator/main.go -v ${TEST_VERSION} -o ./build/test-update ./build/test/state-install
+      $scripts.generate-payload -v ${TEST_VERSION} -o ${TEST_PREPARED_DIR} -i ${TEST_STAGED_DIR}
+
+      go run scripts/ci/update-generator/main.go -v ${TEST_VERSION} -o ${TEST_ARCHIVE_DIR} -i ${TEST_PREPARED_DIR}
```

Comment:
I feel like this is a payload generator on top of a payload generator. The fact that we are now generating this payload with a script should mean that this code should've become much simpler rather than more complex. There shouldn't need to be an interim stage-components dir. It seems the only thing that this is doing that's custom is inject `fakeversioncmd`, so I would suggest running `generate-payload` as normal and then simply injecting the custom cmd at the end.

Reply:
I moved the fake cmd copying into the payload generator as discussed in stand. It works nicely for generate-test-update, and has no effect on generate-update. Tests are passing and generate-test-update is now slimmer.